### PR TITLE
chore(minor): add TimeIntervalCounter, update macos version to use Cl…

### PR DESCRIPTION
**Pull request by blindspotbounty**
TimeIntervalCounter was introduced.
It allows to update external metrics based on time passed using EventTime and any `Clock` protocol object with `now`.
Updated min MacOS version to 13 as `Clock` can be used only with >=v13.

### CODE REVIEW

File: Sources/Helpers/TimeIntervalCounter.swift [hunk 1]
```diff
@@ -0,0 +1,62 @@
+// Copyright 2023 Ordo One AB
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+
```
Comment (dimlio) [inline Sources/Helpers/TimeIntervalCounter.swift:8]: Some description of what this struct is for?
Probably with an example of usage.

Comment (blindspotbounty) [reply Sources/Helpers/TimeIntervalCounter.swift:8]: Added an annotation with description of struct and parameters

File: Sources/Helpers/TimeIntervalCounter.swift [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    /// Sets counter to zero and resets checkpoint time
+    public mutating func reset() {
+        counter = 0
+        nextTimeStamp = clock.now.advanced(by: timeInterval)
+    }
+    
+    /// Increments counter
+    /// returns `true`if timeinterval has been reached with provided time
+    public mutating func inc(eventTime: Time.Instant) -> Bool {
```
Comment (mr-swifter) [inline Sources/Helpers/TimeIntervalCounter.swift:40]: I would prefer to name function with full name: `increment`

File: Sources/Helpers/TimeIntervalCounter.swift [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        counter += 1
+        assert(nextTimeStamp.advanced(by: Time.Duration.zero - timeInterval) < eventTime, "Event time was before time interval")
+        return eventTime > nextTimeStamp
+    }
+    
+    /// Increments counter
+    /// returns `true`if timeinterval has been reached
+    /// uses Time.now as a time point
+    public mutating func inc() -> Bool {
```
Comment (mr-swifter) [inline Sources/Helpers/TimeIntervalCounter.swift:49]: The same here

File: Sources/Helpers/TimeIntervalCounter.swift (hunk 1 above)
Comment (mr-swifter) [reply Sources/Helpers/TimeIntervalCounter.swift:8]: I agree, I think there should be more decoration with example of usage or use case.

**Review by mr-swifter — APPROVED** [review]

File: Sources/Helpers/TimeIntervalCounter.swift (hunk 1 above)
Comment (blindspotbounty) [reply Sources/Helpers/TimeIntervalCounter.swift:8]: Do you mean as a part of documentation or should we use unit tests as example?

Comment (blindspotbounty) [reply Sources/Helpers/TimeIntervalCounter.swift:8]: Btw, should we create a separate case for documentation for `ProcessingRate` as well?

**Pull request merged**